fastplotlib/graphics/vector_field.py

```diff
@@ -0,0 +1,366 @@
+from typing import Sequence
+
+import pygfx
+from pygfx.geometries.utils import merge as merge_geometries
+import pylinalg as la
+import numpy as np
+
+from ._base import Graphic
+from .features import (
+    VectorPositions,
+    VectorDirections,
+)
+
+
+class VectorField(Graphic):
+    _features = {
+        "positions": VectorPositions,
+        "directions": VectorDirections,
+    }
+
+    def __init__(
+        self,
+        positions: np.ndarray | Sequence[float],
+        directions: np.ndarray | Sequence[float],
+        spacing: float,
+        color: str | Sequence[float] | np.ndarray = "w",
+        vector_shape_options: dict = None,
+        size_scaling_factor: float = 1.0,
+        **kwargs,
+    ):
+        """
+        Create a Vector Field. Similar to matplotlib quiver.
+
+        Parameters
+        ----------
+        positions: np.ndarray | Sequence[float]
+            positions of the vectors, array-like, shape must be [n, 2] or [n, 3] where n is the number of vectors.
+
+        directions: np.ndarray | Sequence[float]
+            directions of the vectors, array-like, shape must be [n, 2] or [n, 3] where n is the number of vectors.
+
+        spacing: float
+            average distance between pairs of nearest-neighbor vectors, used for scaling
+
+        color: str | pygfx.Color | Sequence[float] | np.ndarray, default "w"
+            color of the vectors
+
+        vector_shape_options: dict
+            dict with the following fields that describe the shape of the vector arrows.
+            Larger values decrease the size of each component.
+
+                * cone_radius_divisor, default 10.0
+                * cone_height_divisor, default 4.0
+                * stalk_radius_divisor, default 30.0
+                * stalk_height_divisor, default 4.0
```

Comment:
What unit/space are these numbers? Relates to the question about sizing. Maybe a fraction relative to the full length of the vector?